pkg/gitreceive/build.go

```diff
@@ -201,53 +152,72 @@ func build(conf *Config, builderKey, gitSha string) error {
 		return fmt.Errorf("copying %s to %s (%s)", appTgz, tarURL, err)
 	}
 
+	var pod *api.Pod
+	var buildPodName string
+	if usingDockerfile {
+		buildPodName = dockerBuilderPodName(appName, shortSha)
+		pod = dockerBuilderPod(
+			conf.Debug,
+			creds != nil,
+			buildPodName,
+			conf.PodNamespace,
+			"deis",
+			"2.0.0-beta",
+			tarURL,
+			imageName,
+		)
+	} else {
+		buildPodName = slugBuilderPodName(appName, shortSha)
+		pod = slugbuilderPod(
+			conf.Debug,
+			creds != nil,
+			buildPodName,
+			conf.PodNamespace,
+			"deis",
+			"2.0.0-beta",
+			tarURL,
+			pushURL,
+			buildPackURL,
+		)
+	}
 	log.Info("Starting build... but first, coffee!")
 	log.Debug("Starting pod %s", buildPodName)
-	kCreateCmd := exec.Command(
-		"kubectl",
-		fmt.Sprintf("--namespace=%s", conf.PodNamespace),
-		"create",
-		"-f",
-		finalManifestFileLocation,
-	)
-	if log.IsDebugging {
-		kCreateCmd.Stdout = os.Stdout
-	}
-	kCreateCmd.Stderr = os.Stderr
-	if err := run(kCreateCmd); err != nil {
+
+	podsInterface := kubeClient.Pods(conf.PodNamespace)
+
+	newPod, err := podsInterface.Create(pod)
+	if err != nil {
 		return fmt.Errorf("creating builder pod (%s)", err)
 	}
 
-	// poll kubectl every 100ms to determine when the build pod is running
-	// TODO: use the k8s client and watch the event stream instead (https://github.com/deis/builder/issues/65)
-	for {
-		cmd := kGetCmd(conf.PodNamespace, buildPodName)
-		var out bytes.Buffer
-		cmd.Stdout = &out
-		// ignore errors
-		run(cmd)
-		outStr := string(out.Bytes())
-		if strings.Contains(outStr, "phase: Running") {
+	watcher, err := podsInterface.Watch(labels.Nothing(), fields.OneTermEqualSelector("name", pod.Name), v1Version)
```

Comment:
@arschles which version of k8s is being targeted in V2?
I ask because this interface is different in v1.2